src/cli.js

```diff
@@ -122,18 +171,20 @@ function Validator() {
 
     let results = [];
     for (const filename of filenames) {
-      const result = await validateFile(filename, config, plugins, cache);
-      results.push(result);
-
-      for (const plugin of plugins) {
-        const message = plugin.getSingleResultLogMessage(
-          result,
-          filename,
-          config.format,
-        );
-        if (message != null) {
-          logger.log(message);
-          break;
+      const fileResults = await validateFile(filename, config, plugins, cache);
+      results = results.concat(fileResults);
+
+      for (const result of results) {
+        for (const plugin of plugins) {
+          const message = plugin.getSingleResultLogMessage(
```

Comment:
Problem:
The way I'm doing this, this will output

```
✖ ./testfiles/files/multi-doc.yaml[0] is invalid

✔ ./testfiles/files/multi-doc.yaml[1] is valid

✖ ./testfiles/files/multi-doc.yaml[2] is invalid

./testfiles/files/multi-doc.yaml[0]#/num must be number

./testfiles/files/multi-doc.yaml[2]#/num must be number
```
but I want it to output

```
✖ ./testfiles/files/multi-doc.yaml[0] is invalid

./testfiles/files/multi-doc.yaml[0]#/num must be number

✔ ./testfiles/files/multi-doc.yaml[1] is valid

✖ ./testfiles/files/multi-doc.yaml[2] is invalid

./testfiles/files/multi-doc.yaml[2]#/num must be number
```

I need to move things a bout a bit to achieve that